File: iOS/Kalorie/Core/Models/BilingualNamed.swift
//
//  BilingualNamed.swift
//  Kalorie
//
//  Created by Josef Antoni on 24.07.2026.
//

import Foundation

protocol BilingualNamed {
    var czName: String { get }
    var engName: String { get }
}

extension BilingualNamed {
    var displayName: String {
        let langCode = Locale.current.language.languageCode?.identifier ?? ""
        let isCzechOrSlovak = ["cs", "sk"].contains(langCode)
        if isCzechOrSlovak { return czName }
        return engName.isEmpty ? czName : engName
    }
}

File: Android/app/src/main/java/antoni/kalorie/core/models/BilingualNamed.kt
package antoni.kalorie.core.models

import java.util.Locale

interface BilingualNamed {
    val czName: String
    val engName: String
}

val BilingualNamed.displayName: String
    get() {
        val isCzechOrSlovak = Locale.getDefault().language in listOf("cs", "sk")
        if (isCzechOrSlovak) return czName
        return engName.ifEmpty { czName }
    }
